scripts/busybox-ping-test/ping_test.py

```diff
@@ -66,55 +89,79 @@ def start_containers(serv):
+    config_file =  "{}/alcor_services.ini".format(ALCOR_TEST_DIR)
+    config_file_object = read_config_file(config_file)
+    services_dict = dict(config_file_object.items("services"))
+    service_port_map = get_service_port_map(services_dict)
     parser = argparse.ArgumentParser(description='Busybox ping test', epilog='Example of use: python script_name -b')
     parser.add_argument("-b", "--build", type=str, nargs='?', help=' to build alcor services provide :{} as an option'.format('-b build'))
     parser.add_argument("-t", "--testcase", type=int, nargs='?', help='Test case number or {} for all tests cases '.format('all'))
     parser.add_argument("-s", "--all", type=str, nargs='?', help = 'all tests cases')
     args = parser.parse_args()
-    """
+
     if args.build:
         if(args.build == "build"):
-           build_containers(services)
+           build_containers(services_dict)
         else:
-           print("invoke  as {}".format('-b build'))
+           print("To build before running the tests, use '-b build'")
+           print("ERROR: Quitting test\n")
+           sys.exit(1)
 
     stop_containers(service_port_map.keys())
     remove_containers(service_port_map.keys())
-    if(start_containers(services) == True):
+    if(start_containers(services_dict) == True):
       print("All services started Sucessfully")
     else:
-      print("Error:couldn't start all alcor services")
+      print("ERROR: All Alcor services did NOT start successfully")
+      print("ERROR: Quitting test\n")
       sys.exit(1)
-    """
+
+    container_names_dict = dict(config_file_object.items("test_setup"))["container_names"]
+    container_names = json.loads(container_names_dict)
     aca = dict(config_file_object.items("AlcorControlAgents"))
-    aca_nodes_ip_mac = check_alcor_agents_running(aca)
-    if(len(aca_nodes_ip_mac) != len(aca)):
-      print("\nERROR: Alcor Control Agent not running on some of the nodes")
-      print("ERROR: Test exits")
+    for aca_node,con in zip(aca.values(),container_names):
+      print("Busybox container cleanup...", aca_node, con)
+      busybox_container_cleanup(aca_node, con)
+    time.sleep(10)
+
+    check_alcor_agents_running(aca)
+    time.sleep(30)
+
+    if(restart_alcor_agents(aca, ACA_BIN_PATH) == False):
+      print("AlcorControlAgent did NOT start successfully")
+      print("Check the target nodes and run again")
+      print("ERROR: Quitting test\n")
       sys.exit(1)
-    print("Wait for 20 seconds until all services are started...")
-    #time.sleep(20)
+    time.sleep(10)
+
+    aca_nodes_ip_mac = get_macaddr_alcor_agents(aca)
+    print("ACA nodes IP MAC pair::", aca_nodes_ip_mac)
+
+    if len(aca_nodes_ip_mac) != len(aca):
+      print("ERROR: Alcor Control Agent not running on some of the nodes")
+      print("ERROR: Quitting test\n")
+      sys.exit(1)
+
+    print("Wait for 60 seconds until all services are started...")
+    time.sleep(60)
```

Comment:
May be wait in a loop for a set number of times, each time testing if all services are up and running by looking at netstat -ant and count the number of alcor ports that should be in LISTEN mode could be better.

Reply:
We can address this in an upcoming commit.